client/src/components/PreviewIcon.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import styles from './PreviewIcon.module.css'
+
+export const PreviewIcon = ({
+    onClick,
+    previewStyle,
```

Comment:
## Q:
is this for future proofing the Preview So that we can show it in different formats in the future i.e  in a list, as it is currently ect?

Reply:
I'm still not sure whether I will expand this component to become a list item or create another component.
I think I'll be going towards the second option: as you can see in the Explorer component there's a TODO about conditionally displaying a ExplorerList. So my idea was that Explorer handles the data, whereas ExplorerList and ExplorerPreview respectively create ListItem and PreviewIcon to display it.

That attribute you highlighted (previewStyle) it's just there in case we want in the future to have different types of previews. But won't probably be used.